core/ir/riscv64/disassemble.c

```diff
@@ -53,15 +60,34 @@ void
 opnd_base_disp_scale_disassemble(char *buf, size_t bufsz, size_t *sofar INOUT,
                                  opnd_t opnd)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    /* There is no scaled addressing on RISC-V */
+}
+
+static inline const char *
+immed_prefix(void)
+{
+    return (TEST(DR_DISASM_INTEL, DYNAMO_OPTION(disasm_mask))
+                ? ""
+                : (TEST(DR_DISASM_ARM, DYNAMO_OPTION(disasm_mask)) ? "#" : "$"));
 }
 
 bool
 opnd_disassemble_arch(char *buf, size_t bufsz, size_t *sofar INOUT, opnd_t opnd)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    switch (opnd.kind) {
```

Comment:
I thought internal_opnd_disassemble (https://github.com/DynamoRIO/dynamorio/blob/820fa4df87c1f77231c258056ed6a721367c1e6d/core/ir/disassemble_shared.c#L537) needs to call opnd_disassemble_arch only for arch-dependent opnds. But the IMMED_INTEGER being printed here can probably be handled by the existing logic there (except the hex print format probably). Do we really need this separate logic?

Reply:
`internal_opnd_disassemble()` handles negative signed immediate integers only for ARM:
https://github.com/DynamoRIO/dynamorio/blob/820fa4df87c1f77231c258056ed6a721367c1e6d/core/ir/disassemble_shared.c#L546-L552
So I had a choice of either add `RISCV64` to the ifdef in there or put the logic in arch-specific disassemble function.
I preferred the latter because it makes the arch-specific code easier to find. What do you think?

Reply:
Fair enough. Please add a comment by that `#ifdef ARM` saying that we should we move that to arm/disassemble.c.

Reply:
Done.